### DISCUSSION TITLE
Add custom weights

### DISCUSSION THREAD

**Discussion by kumbraGrr** [valhalla/valhalla#3930] [Q&A] [answered]
Hi,
How to add custom cost/weight to specific segment of the road which will be considered as additional factor/cost when performing routes? Is something like this possible?
I am struggling to find answer to this.
I am not referring to the traffic, elevation or excluding road segments in the route request. 
Was thinking about adding extra field in pbf data or something similar that would contain numbers which will be considered as additional cost when performing route request for each segment.
Any info on this is welcomed, 
Thanks

**Comment by nilsnolde** [comment, marked as answer]
Hm no sorry that's not possible. Also not sure I understand how exactly you'd like the API to look like..

It's kind of the whole point of live traffic, so you could bastardize that infrastructure to do that in an external process and Valhalla will respect it. I could also picture `factor_polygons` or `factor_locations` or so that would not exclude, but favor/avoid segments. That would always affect both directions of all edges, but surely has some applications.

**Reply by kumbraGrr** [reply]
Regarding you question I was thinking if there is some option to apply weights like I would with historical traffic data or similar. Csv with applied additional weights and multiplier for the particular segment.
Thank you for the answer, I was thinking to simulate weights with traffic approach. But slight problem is if I have live updates every five minutes and on top of that I would need to multiply additional factor. Still achievable but needs modification. Maybe something to consider for the future. I saw asked feature in some other routing engines. 
Once again, thank you for the answer,
Best regards
